knowledge_repo/app/routes/render.py

```diff
@@ -162,5 +163,28 @@ def about():
 
 
 @blueprint.route('/ajax_post_typeahead', methods=['GET', 'POST'])
-def ajax_post_typehead():
-    return json.dumps(current_app.config.get('typeahead_data', {}))
+def ajax_post_typeahead():
+    # this a string of the search term
+    search_terms = request.args.get('search', '')
+    search_terms = search_terms.split(" ")
+    case_statements = []
+    for term in search_terms:
+        case_stmt = case([(Post.keywords.like('%' + term.strip() + '%'), 1)], else_=0)
```

Comment:
@earthmancash : Changing tags alone (without bumping post revision) already does not trigger an index update. Since tags are embedded in the document, checking every time is expensive. User contributed tags should be read separately, and applied after-the-fact, if we want to search by them at all.